pallets/runtime/src/bridge.rs

```diff
@@ -0,0 +1,303 @@
+//! Bridge from Ethereum to Polymesh
+//!
+//! This module implements a one-way bridge between Polymath Classic on the Ethereum side, and
+//! Polymesh native. It mints POLY on Polymesh in return for permanently locked ERC20 POLY tokens.
+
+use crate::multisig;
+use codec::{Decode, Encode};
+use frame_support::dispatch::{DispatchError, DispatchResult};
+use frame_support::traits::Currency;
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage, ensure, weights::GetDispatchInfo, Parameter,
+};
+use frame_system::{self as system, ensure_signed};
+use polymesh_primitives::{traits::IdentityCurrency, AccountKey, IdentityId, Signatory};
+use polymesh_runtime_balances as balances;
+use polymesh_runtime_common::{balances::Trait as BalancesTrait, CommonTrait};
+use polymesh_runtime_identity as identity;
+use sp_core::H256;
+use sp_runtime::traits::Dispatchable;
+use sp_std::collections::btree_map::BTreeMap;
+use sp_std::{convert::TryFrom, prelude::*};
+
+pub trait Trait: BalancesTrait + multisig::Trait {
+    type Balance: From<u128> + Into<<Self as CommonTrait>::Balance>;
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+    type Proposal: From<Call<Self>>
+        + Into<<Self as identity::Trait>::Proposal>
+        + Parameter
+        + Dispatchable<Origin = Self::Origin>
+        + GetDispatchInfo;
+}
+
+decl_error! {
+    pub enum Error for Module<T: Trait> {
+        /// The validator does not have an identity.
+        IdentityMissing,
+        /// Failure to credit the recipient account.
+        CannotCreditAccount,
+        /// Failure to credit the recipient identity.
+        CannotCreditIdentity,
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Bridge {
+        /// The multisig account of the bridge validator set.
+        Validators get(validators) config(): T::AccountId;
```

Comment:
We can't just set it here without notifying the multisig. We need to do the following actions to make it a valid multisig address
```
for signer in signers.clone() {
                <identity::Module<T>>::add_auth(
                    Signatory::from(AccountKey::try_from(wallet_id.encode())?),
                    signer,
                    AuthorizationData::AddMultiSigSigner,
                    None
                );
            }

            <MultiSigSignsRequired<T>>::insert(&wallet_id, &sigs_required);
```

Reply:
In all cases except at genesis time there should be a valid multisig address. Do you think initialising the validator set address at genesis time is needed? My understanding is that this is a better alternative to requiring the first set of validators to be approved by root.

Reply:
I guess it's a governance decision who gets to add/remove bridge validators. Perhaps it should indeed be the root account. Regardless, as you mentioned, I think it's a good idea to take the initial bridge validators as genesis config input and set these vars accordingly.

Reply:
Rather than changing validators through a Proposal/Call, can't we just directly call the `add_multisig_signer` and `remove_multisig_signer` function of the multisig module with validator's origin?

We can also directly manipulate the `MultiSigSigners` mapping of the multisig module.